server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -198,10 +200,20 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
     public IndexInput openInput(String name, long fileLength, IOContext context) throws IOException {
         InputStream inputStream = null;
         try {
-            inputStream = blobContainer.readBlob(name);
-            return new RemoteIndexInput(name, downloadRateLimiter.apply(inputStream), fileLength);
+            if (context instanceof BlockIOContext) {
```

Comment:
Can we add test for `RemoteDirectory` to verify this path ? In general, I don't see coverage for different changes which this PR is making in new and existing components as well.

Reply:
Added a unit test in RemoteDirectoryTests.java for testing BlockIOContext use case